rasa/shared/exceptions.py

```diff
@@ -13,18 +13,46 @@ class RasaXTermsError(RasaException):
     """Error in case the user didn't accept the Rasa X terms."""
 
 
-class YamlSyntaxException(RasaException):
+class YamlException(RasaException):
+    """Raised if there is an error reading yaml."""
+
+    def __init__(self, filename: Optional[Text] = None):
+        """Create exception.
+
+        Args:
+            filename: optional file the error occurred in"""
+        self.filename = filename
+
+
+class YamlSyntaxException(YamlException):
     """Raised when a YAML file can not be parsed properly due to a syntax error."""
 
-    def __init__(self, filename: Text, underlying_yaml_exception: Exception):
+    def __init__(
+        self,
+        filename: Optional[Text] = None,
+        underlying_yaml_exception: Optional[Exception] = None,
+    ):
         self.underlying_yaml_exception = underlying_yaml_exception
-        self.filename = filename
+        super(YamlSyntaxException, self).__init__(filename)
 
     def __str__(self) -> Text:
-        exception_text = (
-            f"Failed to read '{self.filename}'. " f"{self.underlying_yaml_exception}"
-        )
-        exception_text = exception_text.replace(
-            'in "<unicode string>"', f'in "{self.filename}"'
+        if self.filename:
+            exception_text = f"Failed to read '{self.filename}'."
+        else:
+            exception_text = f"Failed to read YAML."
+
+        if self.underlying_yaml_exception:
+            self.underlying_yaml_exception.warn = None
+            self.underlying_yaml_exception.note = None
+            exception_text += f" {self.underlying_yaml_exception}"
+
+        if self.filename:
+            exception_text = exception_text.replace(
+                'in "<unicode string>"', f'in "{self.filename}"'
+            )
+
+        exception_text += (
+            "\n\nYou can use http://www.yamllint.com/ to validate the "
```

Comment:
(it slightly bugs me out that they are not using https)

Reply:
should we rather link to https://yamlchecker.com/ ?

Reply:
yes please! I also like that it gives instant feedback and I don't have to press `go`. Also it checks for duplicate keys and doesn't transform the yaml in a weird way.